torch/distributed/_tensor/api.py

```diff
@@ -111,8 +114,10 @@ def forward(  # type: ignore[override]
                     input = input.contiguous()
                     device_mesh.broadcast(input, mesh_dim=idx)
 
+        # detach the local tensor passed to DTensor since after the construction
```

Comment:
Again, we should use a view here to say: we want a fresh Tensor object that shares memory and that's it.
There is no autograd concern here.